### PR TITLE
dec inLoop after exiting the while scope in computeLiveRanges [backport]

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

**Comment by Araq** [comment]
This needs to be backported too. CC @narimiran

**Comment by arnetheduck** [comment]
is there a test case for this?

**Comment by ringabout** [comment]
I will make up a testcase soon. The bug should only affect performance and still be correct,

File: tests/arc/tcursorloop.nim [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        echo [it.s]
+        `=copy`(it, it.ri)
+  jt_cursor = root
+  if (
+    not (jt_cursor == nil)):
+    echo [jt_cursor.s]
+    jt_cursor = jt_cursor.ri
+finally:
+  `=destroy`(it)
+-- end of expandArc ------------------------
```
Comment (ringabout) [inline tests/arc/tcursorloop.nim:24]: Hello @arnetheduck Here is the testcase

Without this change `jt` is not optimzied as `jt_cursor`

```nim
--expandArc: traverse

var
  it
  jt
try:
  `=copy`(it, root)
  block :tmp:
    while (
      not (it == nil)):
      if true:
        echo [it.s]
        `=copy`(it, it.ri)
  `=copy`(jt, root)
  if (
    not (jt == nil)):
    echo [jt.s]
    `=copy`(jt, jt.ri)
finally:
  `=destroy`(jt)
  `=destroy`(it)
-- end of expandArc ------------------------
```

**Pull request merged**